### PR TITLE
feat: Docker-compose: add docker compose with App+PostgreSQL

### LINKED ISSUE #72: Create docker-compose backend
Is your feature request related to a problem? Please describe.
It's not a problem

Describe the solution you'd like
We want a docker-compose to be able to run this project easily. The Dockerfile is already in the project

### PR DESCRIPTION
**Pull request by mcastellin**
Creating docker-compose file to spin up a postgresql service with the application.

- Application can build from docker-compose
- Configured Postgresql DB with persistent volume
- DB password can be provided with environment file

Fix #72

### CODE REVIEW

File: docker-compose.yml
Comment (XaviTorello) [inline docker-compose.yml:25]: Current code should be mounted into the container to simplify and enhance development workflow.

It can be achieved with something like:
```
    volumes:
      - .:/anem-per-feina/
```

With this, the base image is not needed to be recreated for each change. Also Djangos reload-on-changes default behavior will be preserved.

File: docker-compose.yml
Comment (XaviTorello) [inline docker-compose.yml:26]: Composition envs should be also loaded here to ensure that same values are passed to both services

```
    env_file:
      - ./docker/secrets.env
```

File: docker/secrets.env.sample
Comment (XaviTorello) [inline docker/secrets.env.sample:1]: Maybe would be clear if we just define an `docker/.env` file to also store or override default vars.

The idea is to load it in our composition to patch "realtime" this values in all related services.

Comment (XaviTorello) [reply docker/secrets.env.sample:1]: I mean, I would also like to see here any composition vars, like `POSTGRES_HOST=db`, `POSTGRES_USER`, ...

What do you think @mcastellin?

File: docker-compose.yml
Comment (XaviTorello) [inline docker-compose.yml:11]: Maybe will be cleaner if we isolate it inside our docker `.env` file

File: Dockerfile
Comment (XaviTorello) [inline Dockerfile:27]: I would not patch .env definition into the image (as a constant), maybe it's better to define it "live" with `env_file` in our composition.

This image should be used also to deploy our service to k8s, CI, ...

**Review by XaviTorello — CHANGES_REQUESTED** [review]
Many thanks for your effort @mcastellin, I just would to share some ideas with you.

Basically I would love to store all the env vars that we'll use to patch the Docker environment in one same place.

Also, I think that is a good idea to mount current code status in our composition to provide an improved development experience without the need to recreate the image for our base container.

I totally loved the integration of `wait-for-it` :)

Thoughts?

File: .dockerignore
Comment (fullonic) [inline .dockerignore:8]: could we add as well `venv`? Is a very common way to name the virtual environment folder.

File: docker-compose.yml
Comment (fullonic) [reply docker-compose.yml:25]: Would be great to the app volume mounted as @XaviTorello suggest,  so we can development already in a docker environment.

File: docker/wait-for-it.sh
Comment (jbagot) [inline docker/wait-for-it.sh:1]: It's fine this. But I normally use `apt install wait-for-it` in the Dockerfile and then you can use it in the docker-compose and you don't need to have this script in the repo.

File: .dockerignore
Comment (jbagot) [reply .dockerignore:8]: yes, agree, `venv` should be ignored as well

File: docker/secrets.env.sample
Comment (mcastellin) [reply docker/secrets.env.sample]: @XaviTorello you can now provide thos variables from the `docker/.env` file

**Comment by mcastellin** [comment]
@XaviTorello @jbagot I have addressed all points to make the docker-compose usable for the development process.
I also modified the dockerfile to use the `slim-buster` version as discussed in #60 if we merge this one, I think the other can be closed since the change is already included here.

I have pushed the code changes and will now also provide some documentation of what you need to do in order to use the image for development.

File: .env.dev.sample
Comment (mcastellin) [inline .env.dev.sample:1]: I have replaced with `#` since the the semicolon was showing warning messages when running the application

File: Dockerfile
Comment (mcastellin) [inline Dockerfile:32]: By default the .env file will not be created or overridden although I have made it possible by using the `DOTENV` build argument in case we want to build the image for staging or production with custom env files.

File: .dockerignore
Comment (fullonic) [reply .dockerignore:8]: @mcastellin Just now I'm seeing that only `db.sqlite3` is being ignored. However the development version of `sqlite` was renamed to `anem-per-feina.sqlite`.

https://github.com/GeeksCAT/anem-per-feina/blob/379c47df481f2aaf8813543dae2011a594777857/.env.dev.sample#L4

Comment (mcastellin) [reply .dockerignore:8]: @fullonic I have updated the ignore statement to include any file ending by `.sqlite3` or `.sqlite`

File: README.md
Comment (jbagot) [inline README.md:66]: typo docker

File: Dockerfile
Comment (jbagot) [reply Dockerfile:32]: For now I'm ok with it. Maybe tomorrow I'll do some modifications on top of it. Great work, thanks.

**Review by jbagot — CHANGES_REQUESTED** [review]

**Comment by jbagot** [comment]
When you try to run the docker-compose doesn't work because the development settings tries to import `.base`. I made a PR solving this bug. Once my PR is merged this docker-compose will work. @mcastellin @XaviTorello

**Comment by jbagot** [comment]
When the typo in the readme will be fixed for me will be approved. After that with this basis I'll do some changes on top of it.

File: .dockerignore
Comment (fullonic) [reply .dockerignore:8]: Great job @mcastellin :)

**Comment by mcastellin** [comment]
Typo corrected, awaiting final review.

@jbagot Thanks for addressing the issue with the `.base` import. I didn't notice that one! 
What is it about the `config-env.sh` that doesn't convince you?

**Comment by jbagot** [comment]
> Typo corrected, awaiting final review.
> 
> @jbagot Thanks for addressing the issue with the `.base` import. I didn't notice that one!
> What is it about the `config-env.sh` that doesn't convince you?

The `config-env.sh` and all the settings approach I think can be easier, I want to merge this PR and then work on that. For example, I would use only one .env, and this .env file has to be there, if the .env doesn't exist, fails. And don't override settings and anything. As simple as read the .env file in the docker-compose. Then in the k8s we will have to think how to manage the secrets and envs. I have couple of ideas but we can discuss it later.
But as I said I'm completely fine with this PR. The only thing is that now @XaviTorello created the pipeline and black is complaining in the `settings/development.py` file. Could you fix it following black please? And then I'll merge it. @mcastellin

File: jobs/settings/development.py
Comment (jbagot) [inline jobs/settings/development.py:4]: Black is complaining because have to be double quote. Run black and it will change everything as it wants.

**Review by jbagot — CHANGES_REQUESTED** [review]

**Comment by mcastellin** [comment]
`development.py` is now formatted. Thanks @jbagot for the explaination

**Review by jbagot — APPROVED** [review]

**Pull request merged**
